code/go/0chain.net/chaincore/transaction/entity.go

```diff
@@ -314,6 +315,7 @@ func (t *Transaction) GetSignatureScheme(ctx context.Context) (encryption.Signat
 	if err != nil {
 		co = client.NewClient()
 		co.ID = t.ClientID
+
 		if err := co.SetPublicKey(t.PublicKey); err != nil {
 			return nil, err
 		}
```

Comment:
We should also check the ClientID and public key here before put it to cache.

Reply:
fixed